onnx/backend/test/case/node/negativeloglikelihoodloss.py

```diff
@@ -220,10 +231,264 @@ def export_input_shape_is_NCd1d2_with_weight_reduction_sum_ignore_index():  # ty
+
+        expect(node, inputs=[input, target], outputs=[negative_log_likelihood_loss],
+            name='test_negative_log_likelihood_loss_input_shape_is_NCd1d2_no_weight_reduction_mean_ignore_index')
+    """
+
+    @staticmethod
+    def export_input_shape_is_NCd1():  # type: () -> None
+        reduction = 'mean'
+        node = onnx.helper.make_node(
+            'NegativeLogLikelihoodLoss',
+            inputs=['input', 'target'],
+            outputs=['loss'],
+            reduction=reduction
+        )
+
+        N, C, d1 = 3, 5, 2
+        np.random.seed(0)
+        input = np.random.rand(N, C, d1).astype(np.float32)
+        target = np.random.randint(0, high=C, size=(N, d1))
+
+        negative_log_likelihood_loss = compute_negative_log_likelihood_loss(input, target, weight=None, reduction=reduction)
+
+        expect(node, inputs=[input, target], outputs=[negative_log_likelihood_loss],
+            name='test_negative_log_likelihood_loss_input_shape_is_NCd1')
+
+    @staticmethod
+    def export_input_shape_is_NCd1_weight():  # type: () -> None
+        reduction = 'mean'
+        node = onnx.helper.make_node(
+            'NegativeLogLikelihoodLoss',
+            inputs=['input', 'target', 'weight'],
+            outputs=['loss'],
+            reduction=reduction
+        )
+
+        N, C, d1 = 3, 5, 2
+        np.random.seed(0)
+        input = np.random.rand(N, C, d1).astype(np.float32)
+        target = np.random.randint(0, high=C, size=(N, d1))
+        weight = np.random.rand(C).astype(np.float32)
+
+        negative_log_likelihood_loss = compute_negative_log_likelihood_loss(input, target, weight=weight, reduction=reduction)
+
+        expect(node, inputs=[input, target, weight], outputs=[negative_log_likelihood_loss],
+            name='test_negative_log_likelihood_loss_input_shape_is_NCd1_weight')
+
+    """
+    TODO : add this test when reference implementation is updated
+    @staticmethod
+    def export_input_shape_is_NCd1_ignore_index():  # type: () -> None
+        reduction = 'mean'
+        ignore_index = np.int64(1)
+        node = onnx.helper.make_node(
+            'NegativeLogLikelihoodLoss',
+            inputs=['input', 'target'],
+            outputs=['loss'],
+            reduction=reduction,
+            ignore_index=ignore_index
+        )
+
+        N, C, d1 = 3, 5, 2
+        np.random.seed(0)
+        input = np.random.rand(N, C, d1).astype(np.float32)
+        target = np.random.randint(0, high=C, size=(N, d1))
+        target[0][0] = np.int64(1)
+
+        negative_log_likelihood_loss = compute_negative_log_likelihood_loss(input, target, weight=None, reduction=reduction, ignore_index=ignore_index)
+
+        expect(node, inputs=[input, target], outputs=[negative_log_likelihood_loss],
+            name='test_negative_log_likelihood_loss_input_shape_is_NCd1_ignore_index')
+    """
```

Comment:
This test assumes ignore_index is within range and if this is failing then we must investigate why function body definition is not working because it should handle cases where ignore_index is within range but if it is not handling that case then we must investigate and fix instead of commenting the test. Please note, the "update of function body" work item is for handling ignore_index that is **out of** range.